cpp/oneapi/dal/algo/svm/backend/cpu/train_cls_kernel.cpp

```diff
@@ -0,0 +1,144 @@
+#include "oneapi/dal/backend/interop/table_conversion.hpp"
+
+#include "oneapi/dal/table/row_accessor.hpp"
+
+namespace oneapi::dal::svm::backend {
+
+using dal::backend::context_cpu;
+using model_t = model<task::classification>;
+using input_t = train_input<task::classification>;
+using result_t = train_result<task::classification>;
+using descriptor_t = detail::descriptor_base<task::classification>;
+
+namespace daal_svm = daal::algorithms::svm;
+namespace daal_classifier = daal::algorithms::classifier;
+namespace interop = dal::backend::interop;
+
+template <typename Float, daal::CpuType Cpu, typename Method>
+using daal_svm_kernel_t =
+    daal_svm::training::internal::SVMTrainImpl<to_daal_method<Method>::value, Float, Cpu>;
+
+template <typename Float, typename Method>
+static result_t call_daal_kernel(const context_cpu& ctx,
+                                 const descriptor_t& desc,
+                                 const table& data,
+                                 const table& labels,
+                                 const table& weights) {
+    const std::int64_t row_count = data.get_row_count();
+    const std::int64_t column_count = data.get_column_count();
+    const std::uint64_t class_count = desc.get_class_count();
+
+    auto arr_label = row_accessor<const Float>{ labels }.pull();
+
+    binary_label_t<Float> unique_label;
+    auto arr_new_label = class_count == 2
+                             ? convert_labels(arr_label, { Float(-1.0), Float(1.0) }, unique_label)
+                             : arr_label;
+    const auto daal_labels = interop::convert_to_daal_homogen_table(arr_new_label, row_count, 1);
+
+    const auto daal_data = interop::convert_to_daal_table<Float>(data);
+    const auto daal_weights = interop::convert_to_daal_table<Float>(weights);
+
+    auto kernel_impl = detail::get_kernel_function_impl(desc);
+    if (!kernel_impl) {
+        throw internal_error{ dal::detail::error_messages::unknown_kernel_function_type() };
+    }
+    const auto daal_kernel = kernel_impl->get_daal_kernel_function();
+
+    const std::uint64_t cache_megabyte = static_cast<std::uint64_t>(desc.get_cache_size());
+    constexpr std::uint64_t megabyte = 1024 * 1024;
+    dal::detail::check_mul_overflow(cache_megabyte, megabyte);
+    const std::uint64_t cache_byte = cache_megabyte * megabyte;
+
+    daal_svm::training::internal::KernelParameter daal_svm_parameter;
+    daal_svm_parameter.kernel = daal_kernel, daal_svm_parameter.C = desc.get_c();
+    daal_svm_parameter.accuracyThreshold = desc.get_accuracy_threshold();
+    daal_svm_parameter.tau = desc.get_tau();
+    daal_svm_parameter.maxIterations =
+        dal::detail::integral_cast<std::size_t>(desc.get_max_iteration_count());
+    daal_svm_parameter.doShrinking = desc.get_shrinking();
+    daal_svm_parameter.cacheSize = cache_byte;
+    daal_svm_parameter.svmType = daal_svm::training::internal::SvmType::CLASSIFICATION;
+
+    if (class_count > 2) {
+        // TODO
```

Comment:
What to-do?

Reply:
This will be implemented as part2: https://github.com/oneapi-src/oneDAL/pull/1462

Reply:
Ok. In the future, it's much better to specify what to-do just in the comment, like '// TODO: blah-blah-blah', because priorities can change and somebody else could follow-up with such a thing.

Reply:
Done, thanks